spec/controllers/provider_foreman_controller_spec.rb

```diff
@@ -271,6 +271,17 @@
     expect(objects).to match_array(expected_objects)
   end
 
+  it "foreman unassigned configuration profile tree node does not list ansible configured systems" do
+    controller.send(:build_configuration_manager_tree, :providers, :configuration_manager_providers_tree)
+    tree_builder = TreeBuilderConfigurationManager.new("root", "", {})
+    unassigned_id = "#{ems_id_for_provider(@provider)}-unassigned"
+    unassigned_configuration_profile = ConfigurationProfile.new(:name       => "Unassigned Profiles Group|#{unassigned_id}",
+                                                                :manager_id => ems_id_for_provider(@provider))
+    objects = tree_builder.send(:x_get_tree_cpf_kids, unassigned_configuration_profile, false)
+    expected_objects = [@configured_system_unprovisioned]
+    expect(objects).to match_array(expected_objects)
```

Comment:
Since it is not explicitly created within this test, I think there needs to be an expectation that there exists an Ansible Tower Configured System that would be displayed elsewhere in the tree.